api/app/signals/apps/api/v1/views/csv.py

```diff
@@ -0,0 +1,68 @@
+import glob
+import os
+
+from django.conf import settings
+from django.http import FileResponse
+from django.utils import timezone
+from rest_framework import renderers, status, viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from signals.apps.api.generics.permissions import SIAPermissions, SIAReportPermissions
+from signals.auth.backend import JWTAuthBackend
+
+
+class PassthroughRenderer(renderers.BaseRenderer):
+    """
+        Return data as-is. View should supply a Response.
+    """
+    media_type = ''
+    format = ''
+
+    def render(self, data, accepted_media_type=None, renderer_context=None):
+        return data
+
+
+class PrivateCsvViewSet(viewsets.ViewSet):
+    """
+    V1 private ViewSet to retrieve generated csv files
+    https://stackoverflow.com/a/51936269
+    """
+
+    authentication_classes = (JWTAuthBackend, )
+    permission_classes = (SIAPermissions & SIAReportPermissions, )
+
+    @action(methods=['get'], detail=True, renderer_classes=(PassthroughRenderer,))
+    def download(self, *args, **kwargs):
+        if not settings.DWH_MEDIA_ROOT:
+            return self._not_found(msg='Unconfigured Csv location')
+
+        now = timezone.now()
+        src_folder = f'{settings.DWH_MEDIA_ROOT}/{now:%Y}/{now:%m}/{now:%d}'
+
+        if not os.path.exists(src_folder):
+            return self._not_found(msg='Incorrect Csv folder')
```

Comment:
Just raise the appropriate exception here and let DRF handle it

Reply:
check